docs/demos.md

```diff
@@ -0,0 +1,29 @@
+# AMT Demos 
+
+These documentations give an overview of how the demos work and explain some mechanism in these demos. (including some functions we use in the demo)
+
+For a better understanding, we suggest that you can try to revise some code to change and verify the behaviour of the demos or dig into them by using commands like `make debug` and trace them one by one.
```

Comment:
I like this but the format is a bit awkward. I'd do something like this to be clearer:

For the best understanding we suggest that you:
* read the document and demo source side by side
* make some changes to the demo to verify its behaviour
* follow it in a debugger with `make debug_<demo>`

Btw I hadn't thought of the last two points, good ideas.

You can link to the debugging doc there too https://github.com/DavidSpickett/ARMMultiTasking/blob/master/docs/debugging.md